streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -251,7 +296,7 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
 
     def stop(graceful: Boolean) {
       // Send the stop signal to all the receivers
-      stopReceivers()
```

Comment:
Its pretty ugly that ReceiverTracker.stop() calls ReceiverLauncher.stop() which issues an event in the  ReceiverTracker.eventLoop, which calls ReceiverTracker.stopReceivers(). This should be collapsed. I think the ReceiverLAuncher.stop can be removed and all that code be moved into ReceiverTracker.stop(). So ...
1. ReceiverTracker.stop() should issue the `StopAllReceivers` event, and wait for `running` = false 
2. The event loop, on `StopAllReceivers` event, will call the method stopReceivers() to send the signal (as it is now).